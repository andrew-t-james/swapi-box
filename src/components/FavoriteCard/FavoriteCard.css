.favorite__card {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.8);
  display: flex;
  font-size: 1.2rem;
  font-weight: 100;
  height: 5rem;
  justify-content: space-between;
  margin: 0 auto;
  padding: 1rem 0;
  width: 20rem;
}

.favorite__card--button {
  outline: none;
}