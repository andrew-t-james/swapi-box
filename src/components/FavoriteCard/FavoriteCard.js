import React from 'react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

const FavoriteCard = props => {
  return (
    <div>
      <h2>{props.name}</h2>
      <span>
        <Button name="⭐" />
      </span>
    </div>
  );
};

FavoriteCard.propTypes = {

};

export default FavoriteCard;
