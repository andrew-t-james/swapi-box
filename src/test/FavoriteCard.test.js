import React from 'react';
import { shallow } from 'enzyme';
import FavoriteCard from '../components/FavoriteCard/FavoriteCard';

describe('FavoriteCard', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<FavoriteCard />));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
