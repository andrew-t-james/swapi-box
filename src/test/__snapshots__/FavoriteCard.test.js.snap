// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <span>
          <Button
                    name="⭐"
          />
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            name="⭐"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "⭐",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <span>
            <Button
                        name="⭐"
            />
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              name="⭐"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "⭐",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
